fastlane/lib/fastlane/actions/ensure_git_status_clean.rb

```diff
@@ -23,7 +23,25 @@ def self.run(params)
         # Manual post processing trying to ignore certain file paths
         if (ignore_files = params[:ignore_files])
           repo_status = repo_status.lines.reject do |line|
-            path = line.split(' ').last
+            # Extract the file path from the line based on
+            # https://git-scm.com/docs/git-status#_output
+
+            # The first two characters indicate the status of the file path (e.g. ' M')
+            # XY PATH
+            path = line&.match(/^.{2}\s(.*)/)&.[](1) || ''
+
+            # If the file path is renamed, the original path is also included in the line (e.g. ' R ORIG_PATH -> PATH')
+            # XY ORIG_PATH -> PATH
+            path = path.split('->').last.strip if path.include?('->')
+
+            # Remove double quotes from the file path
+            path = path.gsub(/"/, '')
```

Comment:
this could be risky if the file name contained a quote.

Maybe you can solve the whole matching using a single regular expression

```ruby
def path_from_git_status_line(line)
  path = line.match(/^.. (.* -> )?(.*)$/)[2]
  path = path.delete_prefix('"').delete_suffix('"')
end

lines=[ 
 ' M .previous_issue_count',
 ' M "A File Path"',
 ' M .previous_issue_count -> .previous_issue_count.new',
 ' M "A/File Path" -> "A/File Path.new"']

lines.map{|l| path_from_git_status_line(l)}
```

https://rubular.com/

Reply:
`git status --porcelain` escapes the quote and the output looks like this: 

```
 M "\"quote.swift"
```

To ignore a file with a quote in the name, users will need to espace it in the Fastfile too:

```
ensure_git_status_clean(
  ignore_files: ["\\\"quote.swift"]
)
```

Thanks for the suggestion! Fixed ✔️